opentelemetry-sdk/tests/metrics/test_instrument.py

```diff
@@ -324,6 +325,30 @@ def test_record(self):
         hist.record(1.0)
         mc.consume_measurement.assert_called_once()
 
+    def test_time(self):
+        mc = Mock()
+        hist = Histogram("name", Mock(), mc)
+
+        @hist.time()
+        def foo():
+            time.sleep(0.5)
+
+        foo()
+        measurement = mc.consume_measurement.call_args[0][0]
+        mc.consume_measurement.assert_called_once()
+        self.assertGreaterEqual(measurement.value, 500)
+
+    def test_time_attributes_update(self):
+        mc = Mock()
+        hist = Histogram("name", Mock(), mc)
+        with hist.time() as measurer:
+            time.sleep(0.2)
+            measurer.set_attributes({"k": "v"})
```

Comment:
Maybe this is be more clear, here it is shown that the user can associate repeatedly more attributes to the time measurement since `set_attributes` can be called more than once. Those attributes should be set once when the `time` method is called instead.

Reply:
I added this since it's not possible to get all the attributes upfront in many cases such as status_code until the request is complete.

Reply:
Ok, I understand.